packages/utils/src/index.ts

```diff
@@ -5,6 +5,67 @@ export type UndefinedToOptional<T extends object> = Partial<T> &
 // https://stackoverflow.com/a/51390763
 export type Falsy = false | 0 | '' | null | undefined
 
+export type ReadonlyDeep<T> = T extends BuiltIns
```

Comment:
Not sure it is even needed, looks like extra complexity and place for additional bugs